docs/modules/ROOT/pages/documentation/payara-server/server-configuration/security/certificate-realm-certificate-validation.adoc

```diff
@@ -0,0 +1,53 @@
+[[client-certificate-validator]]
+= Client Certificate Custom Validations
+
+_Since Payara Server 5.2021.5_
+
+
+By default, Payara checks only if the submitted Client Certificate is present in the trust store. If it is found, it is accepted without any additional checks.
+
+Since Payara 5.2021.5, you can now implement a Client Certificate Validator which performs additional checks in addition to the presence of the certificate in the trust store.  You can perform a check on the validity date, use the Online Certificate Status Protocol (OCSP) to validate the certificate when it has a Certificate Revocation List entry (CRL), or lookup the DN name in a database for example, and determine based on this information if the certificate is still accepted.
```

Comment:
```suggestion
Custom Client Certificate Validation allows you to perform extra checks in addition to checking for the presence of the certificate in the trust store.
You can perform a check on the validity date, and use the Online Certificate Status Protocol (OCSP) to validate the certificate when it has a Certificate Revocation List entry (CRL), or lookup the DN name in a database for example, and determine based on this information if the certificate is still accepted.
```

Reply:
There is no need to include the specific version which the version has been introduced in; it has been decided users can have a look through release notes if they're interested